src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -288,6 +304,69 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
         }, () => this.canvas.current.applyTag(tag.name));
     }
 
+    /**
+     * Open confirm dialog for tag renaming
+     */
+    private confirmTagRenamed = (tagName: string, newTagName: string): void => {
+        this.renameTagConfirm.current.open(tagName, newTagName);
+    }
+
+    /**
+     * Renames tag in assets and project, and saves files
+     * @param tagName Name of tag to be renamed
+     * @param newTagName New name of tag
+     */
+    private onTagRenamed = async (tagName: string, newTagName: string): Promise<void> => {
+        const { project, selectedAsset } = this.state;
+        const newProject = {
+            ...project,
+            tags: project.tags.map((t) => (t.name === tagName) ? {...t, name: newTagName} : t),
+        };
+
+        const assetService = new AssetService(newProject);
+        const asset = await assetService.renameTag(project.assets, tagName, newTagName, selectedAsset);
```

Comment:
`AssetService` already has a reference to a project (which has the assets).  Any reason to still pass the assets here?

Reply:
Also seems a bit strange that this takes / and returns the selected asset?

Reply:
1. Kinda forgot about that. Good call
2. I was going to ask you about that. I was just looking for a good way to refresh the current selected asset. It seemed like it would be less efficient to try and load it from the file again since I was going to use it right there. Open to suggestions